src/main/java/gregtech/common/tileentities/machines/basic/GT_MetaTileEntity_BasicDrillerBase.java

```diff
@@ -0,0 +1,138 @@
+package gregtech.common.tileentities.machines.basic;
+
+import gregtech.api.interfaces.ITexture;
+import gregtech.api.interfaces.tileentity.IGregTechTileEntity;
+import gregtech.api.metatileentity.implementations.GT_MetaTileEntity_BasicMachine;
+import gregtech.api.util.GT_ModHandler;
+import gregtech.api.util.GT_Utility;
+import net.minecraft.block.Block;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraftforge.common.util.FakePlayer;
+import org.lwjgl.Sys;
+
+import static gregtech.api.enums.GT_Values.V;
+
+public abstract class GT_MetaTileEntity_BasicDrillerBase extends GT_MetaTileEntity_BasicMachine {
+
+    static int[] RADIUS, //Miner radius per tier
+    SPEED, //Miner cycle time per tier
+    ENERGY ; //Miner energy consumption per tier
+
+    protected static final ItemStack MINING_PIPE = GT_ModHandler.getIC2Item("miningPipe", 0);
+    protected static final Block MINING_PIPE_BLOCK = GT_Utility.getBlockFromStack(MINING_PIPE);
+    protected static final Block MINING_PIPE_TIP_BLOCK = GT_Utility.getBlockFromStack(GT_ModHandler.getIC2Item("miningPipeTip", 0));
+
+    int drillX, drillY, drillZ;
+    boolean isPickingPipes;
+    boolean waitMiningPipe = true;
+
+    public GT_MetaTileEntity_BasicDrillerBase(int aID, String aName, String aNameRegional, int aTier, String[] aDescription, int aInSlots, int aOutSlots, String aGUIName, ITexture... aTextures){
+        super(aID,aName,aNameRegional,aTier,1,aDescription,aInSlots,aOutSlots,aGUIName,"",aTextures);
+    }
+
+    public GT_MetaTileEntity_BasicDrillerBase(String aName, int aTier, String[] aDescription, ITexture[][][] aTextures, int aInputSlotCount, int aOutputSlotCount, String aGUIName) {
+        super(aName,aTier,1,aDescription,aTextures,aInputSlotCount,aOutputSlotCount,aGUIName,"");
+    }
+
+    public boolean allowPutStack(IGregTechTileEntity aBaseMetaTileEntity, int aIndex, byte aSide, ItemStack aStack) {
+        return (super.allowPutStack(aBaseMetaTileEntity, aIndex, aSide, aStack)) && (aStack.getItem() == MINING_PIPE.getItem());
+    }
+
+    public abstract boolean hasFreeSpace();
+
+    @Override
+    public void onPostTick(IGregTechTileEntity aBaseMetaTileEntity, long aTick) {
+        super.onPostTick(aBaseMetaTileEntity, aTick);
+        if (aBaseMetaTileEntity.isServerSide()) {
+            if (aBaseMetaTileEntity.isAllowedToWork() && aBaseMetaTileEntity.isUniversalEnergyStored(ENERGY[mTier] * (SPEED[mTier] - mProgresstime)) && hasFreeSpace()) {
```

Comment:
Было бы круто вынести в отдельный метод проверки связанные с TE
```suggestion
            if (
                isTEAllowedToWorkAndStoredEnoughEnergy(aBaseMetaTileEntity)
                && hasFreeSpace()
            ) {
            
            ...
            private bool isTEAllowedToWorkAndStoredEnoughEnergy(IGregTechTileEntity aBaseMetaTileEntity)
            {
                return aBaseMetaTileEntity.isAllowedToWork() && aBaseMetaTileEntity.isUniversalEnergyStored(ENERGY[mTier] * (SPEED[mTier] - mProgresstime));
            }
```

Reply:
нет ни малейшего смысла создавать функции которые будут использованы ровно 1 раз, функции нужны для упрощения понимания или для уменьшения количества кода, тут имхо ни того, ни другого

Reply:
К сожалению, с этой стороны есть только мое имхо по которому и я предложил данное изменение.